src/Microsoft.DotNet.Interactive.AspNetCore/AspNetCoreKernelExtensions.cs

```diff
@@ -0,0 +1,104 @@
+        {
+            typeof(Host).Assembly, // Microsoft.Extensions.Hosting
+            typeof(WebHost).Assembly, // Microsoft.AspNetCore
+            typeof(Controller).Assembly, // Microsoft.AspNetCore.Mvc.ViewFeatures
+            typeof(DeveloperExceptionPageMiddleware).Assembly, // Microsoft.AspNetCore.Diagnostics
+            typeof(AspNetCoreKernelExtensions).Assembly, // Microsoft.DotNet.Interactive.AspNetCore
+        };
+
+        public static CSharpKernel UseAspNetCore(this CSharpKernel kernel)
+        {
+            var isActive = false;
+
+            var directive = new Command("#!aspnet", "Activate ASP.NET Core")
+            {
+                Handler = CommandHandler.Create<string, string, KernelInvocationContext>(async (from, name, context) =>
+                {
+                    if (isActive)
+                    {
+                        return;
+                    }
+
+                    isActive = true;
+
+                    // We could try to manage lifetime, but for now just stop the kernel if you want to stop the host.
+                    var interactiveHost = new InteractiveHost();
+                    var startHostTask = interactiveHost.StartAsync();
+
+                    var rDirectives = string.Join(Environment.NewLine, _references.Select(a => $"#r \"{a.Location}\""));
+                    var usings = string.Join(Environment.NewLine, _namespaces.Select(ns => $"using {ns};"));
+
+                    await kernel.SendAsync(new SubmitCode($"{rDirectives}{Environment.NewLine}{usings}"), CancellationToken.None);
+                    await startHostTask;
+
+                    await kernel.SetVariableAsync<IApplicationBuilder>("App", interactiveHost.App);
+                    await kernel.SetVariableAsync<IEndpointRouteBuilder>("Endpoints", interactiveHost.Endpoints);
+                    await kernel.SetVariableAsync("HttpClient", HttpClientFormatter.CreateEnhancedHttpClient(interactiveHost.Address));
+                })
+            };
+
+            kernel.AddDirective(directive);
+
+            Formatter.Register<HttpResponseMessage>((responseMessage, textWriter) =>
+            {
+                // Formatter.Register() doesn't support async formatters yet.
+                // Prevent SynchronizationContext-induced deadlocks given the following sync-over-async code.
+                ExecutionContext.SuppressFlow();
+
+                try
+                {
+                    HttpClientFormatter.FormatHttpResponseMessage(responseMessage, textWriter).Wait();
+                }
+                finally
+                {
+                    ExecutionContext.RestoreFlow();
+                }
+            }, HtmlFormatter.MimeType);
+
+            return kernel;
+        }
+
+        // REVIEW: Should we make this a public CSharp/DotNetKernel extension method?
+        private static Task SetVariableAsync<T>(this CSharpKernel kernel, string name, T value)
```

Comment:
Should we make this public?

Reply:
I think a public version of this is useful but should be on `DotNetKernel` (or ` T where T : DotNetKernel`) rather than on `CSharpKernel`.